core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -315,11 +325,17 @@ else if (isolationType == FetchIsolation.HIGH_WATERMARK)
     }
 
     private Map<TopicIdPartition, LogReadResult> readFromLog(Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData) {
+        // Filter if there already exists any erroneous topic partition.
+        Set<TopicIdPartition> partitionsToFetch = shareFetch.filterErroneousTopicPartitions(topicPartitionData.keySet());
```

Comment:
It's probably better to do this in `acquirablePartitions()` since it avoids acquiring the locks for error partitions.

Reply:
The getPartitionorException API is called after acquiring share partitions hence if some share partition has errored out after acquiring then the readFromLog will trigger for that topicPartition. Hence it's best to filter before the expecsive `readFromLog` API call.
Also if the share partition is errored which is not recoverable then in `acquirablePartitions` the fetch lock API should return false.